flink-connectors/flink-connector-files/src/main/java/org/apache/flink/connector/file/table/utils/PathUtils.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.file.table.utils;
+
+import org.apache.flink.core.fs.FileSystem;
+import org.apache.flink.core.fs.Path;
+import org.apache.flink.util.Preconditions;
+
+import java.io.IOException;
+import java.util.UUID;
+
+/** Path utils for file system. */
+public class PathUtils {
```

Comment:
I originally planned to move the logic for generating the staging directory into `FileSystemOutputFormat`. However, when modifying the `HiveTableSink`, I found that `createBatchCompactSink` relies on tmpPath.

I didn't want to introduce changes to other modules, so I ultimately opted for a compromise and extracted this utility method.

Reply:
`HiveTableSink#createBatchCompactSink` uses the random staging directory for the [BatchFileWriter instantiation in [line 490ff](https://github.com/apache/flink/blob/06caf191991b944769cb7b02769b5d769064febf/flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveTableSink.java#L492). The `BatchFileWriter` instantiates a `PartitionTempFileManager` in the operator's open method ([BatchFileWriter:85](https://github.com/apache/flink/blob/0f3470db83c1fddba9ac9a7299b1e61baab4ff12/flink-connectors/flink-connector-files/src/main/java/org/apache/flink/connector/file/table/batch/compact/BatchFileWriter.java#L85)). The `PartitionTempFileManager` creates a subfolder for the attempt and deletes any existing directory to ensure that no side effects occur.

That said, the two code locations seem to be independent from each other. Moving the staging directory into the OutputFormat still sounds like a reasonable thing to do. But that's your call in the end.